src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -6609,6 +6610,25 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
   ENSURE_NO_EVAL_ERROR
   bool cacheEnabled = cacheValue.toBool();
 
+  // Sixth parameter (for intersects only) is the min overlap (area or length)
+  // Seventh parameter (for intersects only) is the min inscribed circle radius
+  double minOverlap { -1 };
+  double minInscribedCircleRadius { -1 };
+  if ( isIntersectsFunc )
+  {
+
+    node = QgsExpressionUtils::getNode( values.at( 5 ), parent ); //in expressions overlay functions throw the exception: Eval Error: Cannot convert '' to int
```

Comment:
why can't we just use QgsExpressionUtils::getDoubleValue directly here?

Reply:
I think it's because the valued could be an expression, I took the same approach from a few lines above: https://github.com/qgis/QGIS/pull/46185/files#diff-1edfa3cc15da06ba954e37e8662a5bf8ad502e5e0c3b10b055e111bbaa83d18eL6598

Reply:
> I think it's because the valued could be an expression

That shouldn't matter, the expression engine will already handle that nicely. Does it work if you use getDoubleValue? (I suspect the lines above this using the long way around are likely incorrect)

Reply:
Ok, I tried and it doesn't work, now I'll get to understand why...

Reply:
``c++
minOverlap = QgsExpressionUtils::getDoubleValue( values.at( 5 ), parent );
```

this doesn't work because `getDoubleValue()` does not work with expression nodes and `values.at( 5 )` returns an expression node (wrapped into a `QVariant`).

![immagine](https://user-images.githubusercontent.com/142164/143685057-0c1da3a4-6565-4860-ae67-7a90d8f272c8.png)